### PR TITLE
catppuccin-gtk: 0.7.5 -> 1.0.3

### PR DESCRIPTION
**Pull request by isabelroses**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ca/catppuccin-gtk/package.nix [hunk 1]
```diff
@@ -20,73 +19,61 @@ let
...
-  patches = [
-    ./colloid-src-git-reset.patch
-  ];
-
-  buildInputs = [
-    gnome-themes-extra
+  nativeBuildInputs = [
+    gtk3
+    sassc
+    git
```
Comment (Aleksanaa) [inline pkgs/by-name/ca/catppuccin-gtk/package.nix:43]: Git? There's no .git in source

Comment (isabelroses) [reply pkgs/by-name/ca/catppuccin-gtk/package.nix:43]: They are applying patches to colloid via `git apply` https://github.com/catppuccin/gtk/blob/4173b70b910bbb3a42ef0e329b3e98d53cef3350/build.py#L465

Comment (Aleksanaa) [reply pkgs/by-name/ca/catppuccin-gtk/package.nix:43]: Oh okay, maybe leave a comment there as this is usually seen in some wrong practices

Comment (isabelroses) [reply pkgs/by-name/ca/catppuccin-gtk/package.nix:43]: Sure i can.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -421,8 +421,6 @@ with pkgs;
 
   catppuccin-catwalk = callPackage ../development/tools/misc/catppuccin-catwalk { };
 
-  catppuccin-gtk = callPackage ../data/themes/catppuccin-gtk { };
-
```
Comment (dixslyf) [inline pkgs/top-level/all-packages.nix:425]: Is this supposed to be removed?

Comment (dixslyf) [reply pkgs/top-level/all-packages.nix:425]: Ah, got it. I didn't notice the migration to `pkgs/by-name`.

**Review by dixslyf — APPROVED** [review]
Tested with my configuration. LGTM.

Result of `nixpkgs-review pr 315143` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>catppuccin-gtk</li>
  </ul>
</details>

**Review by dixslyf — APPROVED** [review]
Result of `nixpkgs-review pr 315143` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>catppuccin-gtk</li>
  </ul>
</details>

**Comment by isabelroses** [comment]
Force push fixed merge conflict.

**Review by dixslyf — APPROVED** [review]
Result of `nixpkgs-review pr 315143` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>catppuccin-gtk</li>
  </ul>
</details>

**Comment by Aleksanaa** [comment]
I don't fully understand the problems encountered by catppuccin-gtk. Do versions after 1.0 have more regressions than previous ones? Should we drop this package and prefer colloid (since [Vince has added catppuccin flavor](https://github.com/vinceliuice/Colloid-gtk-theme/issues/61#issuecomment-2170030302))?

**Comment by isabelroses** [comment]
> I don't fully understand the problems encountered by catppuccin-gtk. 

Largely issues that simply shouldn't exist that pushed the maintainers over the edge.

> Do versions after 1.0 have more regressions than previous ones? 

They should function similarly with some changes, if not better. But are more broken on some distros then others, from personal testing works fine on nixos.

> Should we drop this package and prefer colloid (since [Vince has added catppuccin flavor](https://github.com/vinceliuice/Colloid-gtk-theme/issues/61#issuecomment-2170030302))?

This does not seem like a bad idea to be honest.

**Review by Aleksanaa — APPROVED** [review]

**Comment by Kranzes** [comment]
What's the point of this package now if it doesn't produce any themes in `$out`?
```
[kranzes@pongo ~/.cache/nixpkgs-review/pr-315143/results/catppuccin-gtk]$ tree
.
└── share
    └── themes

3 directories, 0 files
```

**Comment by khaneliman** [comment]
> What's the point of this package now if it doesn't produce any themes in `$out`?
> 
> ```
> [kranzes@pongo ~/.cache/nixpkgs-review/pr-315143/results/catppuccin-gtk]$ tree
> .
> └── share
>     └── themes
> 
> 3 directories, 0 files
> ```
> 
> Edit: It seems there was a build failure, why was this merged?
> 
> https://hydra.nixos.org/build/263568619/log

Yeah this PR seems to have broke catppuccin gtk theming for me..

**Comment by Aleksanaa** [comment]
```
[catppuccin-gtk] [ERROR] - Something went wrong when building the theme:
Traceback (most recent call last):
  File "/build/source/build.py", line 16, in <module>
    execute_build(git_root, args)
  File "/build/source/sources/build/__init__.py", line 25, in execute_build
    apply_colloid_patches(colloid_tmp_dir, patch_dir)
  File "/build/source/sources/build/patches.py", line 28, in apply_colloid_patches
    subprocess.check_call(
  File "/nix/store/6b1fqdwb3g56j5pazv8zkx9qd0mv3wiz-python3-3.11.9/lib/python3.11/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
```

This didn't happen on ofborg:

```
[catppuccin-gtk] [INFO] - Applying patches...
[catppuccin-gtk] [INFO] - Applying patch 'plank-dark.patch', located at 'sources/patches/colloid/plank-dark.patch'
[catppuccin-gtk] [INFO] - Applying patch 'plank-light.patch', located at 'sources/patches/colloid/plank-light.patch'
[catppuccin-gtk] [INFO] - Applying patch 'sass-palette-frappe.patch', located at 'sources/patches/colloid/sass-palette-frappe.patch'
[catppuccin-gtk] [INFO] - Applying patch 'sass-palette-mocha.patch', located at 'sources/patches/colloid/sass-palette-mocha.patch'
[catppuccin-gtk] [INFO] - Applying patch 'sass-palette-latte.patch', located at 'sources/patches/colloid/sass-palette-latte.patch'
[catppuccin-gtk] [INFO] - Applying patch 'sass-palette-macchiato.patch', located at 'sources/patches/colloid/sass-palette-macchiato.patch'
[catppuccin-gtk] [INFO] - Applying patch 'theme-func.patch', located at 'sources/patches/colloid/theme-func.patch'
[catppuccin-gtk] [INFO] - Patching finished.
[catppuccin-gtk] [INFO] - Building temp tweaks file
[catppuccin-gtk] [INFO] - Inserting gnome-shell imports
[catppuccin-gtk] [INFO] - Building main theme
[catppuccin-gtk] [INFO] - Build info:
    build_root: /nix/store/cm9yzv5q8vpfbzad8m1j7a4rk17a5kdw-catppuccin-gtk-1.0.3/share/themes
    src_root:   /build/source/.tmp/colloid-tmp-frappe/src
    theme_name: catppuccin
    flavor:     frappe
    accent:     blue
    size:       standard
    tweaks:     Tweaks(tweaks=[])
[catppuccin-gtk] [INFO] - Building into '/nix/store/cm9yzv5q8vpfbzad8m1j7a4rk17a5kdw-catppuccin-gtk-1.0.3/share/themes/catppuccin-frappe-blue-standard+default'...
[catppuccin-gtk] [INFO] - Main build complete
[catppuccin-gtk] [INFO] - Bundling assets...
[catppuccin-gtk] [INFO] - Asset bundling done
[catppuccin-gtk] [INFO] - Completed frappe with blue
[catppuccin-gtk] [INFO] - Done!
[catppuccin-gtk] [INFO] - 
[catppuccin-gtk] [INFO] - Built in 1.246s
```

**Comment by Aleksanaa** [comment]
Oh...

My fault 😨 -> Not my fault 🙃 -> Still my fault 🫠

I updated colloid.

**Comment by Aleksanaa** [comment]
A quick fix: try out `colloid-gtk-theme.override { tweaks = [ "catppuccin" ]; }` ...

**Comment by nonetrix** [comment]
Yep it brokee linux ricing is my passion
image

**Comment by stelcodes** [comment]
This broke my theming too just like @nonetrix 🥺

**Comment by isabelroses** [comment]
> This broke my theming too just like @nonetrix 🥺

This should have been fixed by now. Try making the theme name all lower case, and remove `-Dark` and if you have any tweaks do `+tweak,other-tweak`

**Comment by nonetrix** [comment]
Everything works for me now can confirm, and I think gimp is the only thing that uses GTK2 still and it's fine doesn't apply theme though ¯\_(ツ)_/¯

**Comment by stelcodes** [comment]
I'm just gonna stick to 0.7.5 going forward from a copy pasta derivation. The newer version has subtle differences (like waybar button hover styling?) that I don't like. x(

**Comment by nonetrix** [comment]
In Waybar specifically? Just edit the CSS for it inside of Waybar

**Pull request merged**